tests/api/test_collect_runs_logs.py

```diff
@@ -0,0 +1,211 @@
+        )
+
+        await mlrun.api.main._initiate_logs_collection(self.start_log_limit)
+
+        assert (
+            mlrun.api.utils.singletons.db.get_db().update_runs_requested_logs.call_count
+            == 1
+        )
+        assert (
+            mlrun.api.utils.singletons.db.get_db().update_runs_requested_logs.call_args[
+                1
+            ]["uids"]
+            == run_uids
+        )
+
+    @pytest.mark.asyncio
+    async def test_collect_logs_with_no_runs(
+        self,
+        db: sqlalchemy.orm.session.Session,
+        client: fastapi.testclient.TestClient,
+    ):
+        log_collector = mlrun.api.utils.clients.log_collector.LogCollectorClient()
+
+        runs = mlrun.api.utils.singletons.db.get_db().list_distinct_runs_uids(
+            db,
+            requested_logs=False,
+            only_uids=False,
+        )
+        assert len(runs) == 0
+
+        log_collector._call = unittest.mock.AsyncMock(
+            return_value=StartLogResponse(True, "")
+        )
+        mlrun.api.utils.singletons.db.get_db().update_runs_requested_logs = (
+            unittest.mock.Mock()
+        )
+
+        await mlrun.api.main._initiate_logs_collection(self.start_log_limit)
+
+        assert (
+            mlrun.api.utils.singletons.db.get_db().update_runs_requested_logs.call_count
+            == 0
+        )
+
+    @pytest.mark.asyncio
+    async def test_collect_logs_with_runs_fails(
+        self, db: sqlalchemy.orm.session.Session, client: fastapi.testclient.TestClient
+    ):
+        log_collector = mlrun.api.utils.clients.log_collector.LogCollectorClient()
+
+        project_name = "some-project"
+        run_uids = ["some_uid", "some_uid2", "some_uid3"]
+        for run_uid in run_uids:
+            for i in range(3):
+                _create_new_run(
+                    db, project_name, uid=run_uid, name=run_uid, iteration=i, kind="job"
+                )
+
+        runs = mlrun.api.utils.singletons.db.get_db().list_distinct_runs_uids(
+            db,
+            requested_logs=False,
+            only_uids=False,
+        )
+        assert len(runs) == 3
+
+        log_collector._call = unittest.mock.AsyncMock(
+            return_value=StartLogResponse(False, "some error")
+        )
+        mlrun.api.utils.singletons.db.get_db().update_runs_requested_logs = (
+            unittest.mock.Mock()
+        )
+
+        await mlrun.api.main._initiate_logs_collection(self.start_log_limit)
+
+        assert (
+            mlrun.api.utils.singletons.db.get_db().update_runs_requested_logs.call_count
+            == 0
+        )
+
+    @pytest.mark.asyncio
+    async def test_start_log_for_run_success_local_kind(
+        self, db: sqlalchemy.orm.session.Session, client: fastapi.testclient.TestClient
+    ):
+        log_collector = mlrun.api.utils.clients.log_collector.LogCollectorClient()
+        log_collector._call = unittest.mock.AsyncMock(
+            return_value=StartLogResponse(True, "")
+        )
```

Comment:
You do this every test, maybe turn this into a fixture that accepts `success, error_str` as params?